src/blr/cli/tagfastq.py

```diff
@@ -0,0 +1,132 @@
+            name_and_pos_r2, read_and_index_r2 = read2.name.split(maxsplit=1)
+
+            assert name_and_pos_r1 == name_and_pos_r2
+
+            try:
+                raw_barcode_seq = header_barcodes[name_and_pos_r1]
+            except KeyError:
+                reads_skipped.append(name_and_pos_r1)
+                continue
+
+            corr_barcode = corrected_barcodes[raw_barcode_seq]
+
+            # Make new string with barcode sequence and id to add to headers.
+            new_text = f"{corr_barcode.seq}_{args.barcode_cluster_tag}:Z:{corr_barcode.id}"
+
+            # Save header to read instances
+            read1.name = f"{name_and_pos_r1}_{new_text} {read_and_index_r1}"
+            read2.name = f"{name_and_pos_r2}_{new_text} {read_and_index_r2}"
+
+            # Write to out
+            writer.write(read1, read2)
+
+    for header in reads_skipped:
+        logger.info(f"Barcode for read-pair {header} not found. Skipped read-pair.")
+
+    logger.info(f'Finished')
+
+
+def parse_corrected_barcodes(open_file):
+    """
+    Parse starcode cluster output and return a dictionary with raw sequences pointing to a
+    corrected canonical sequence
+    :param open_file: starcode tabular output file.
+    :return: dict: raw sequences pointing to a namedtuple containing corrected canonical sequence and id.
+    """
+    target = namedtuple("Cluster", ['seq', 'id'])
+    corrected_barcodes = dict()
+    for id, cluster in tqdm(enumerate(open_file.readlines(), start=1), desc="Corrected barcodes processed"):
+        canonical_seq, _, cluster_seqs = cluster.strip().split("\t", maxsplit=3)
+        cluster_target = target(seq=canonical_seq, id=id)
+        corrected_barcodes.update({raw_seq: cluster_target for raw_seq in cluster_seqs.split(",")})
+    return corrected_barcodes
+
+
+def parser_raw_barcodes(open_file):
+    """
+    Parse FASTA/FASTQ containing barcode sequences and return a dictionary with entry headers pointing to a
+    raw barcodes sequence.
+    :param open_file: dnaio odject.
+    :return: dict: entry headers pointing to a raw barcodes sequence
+    """
+    header_barcodes = dict()
+    for barcode in tqdm(open_file, desc="Raw barcodes processed"):
+        header, _ = barcode.name.split(maxsplit=1)
+        header_barcodes[header] = barcode.sequence
+    return header_barcodes
+
+
+def add_arguments(parser):
+    parser.add_argument(
+        "input1", metavar='<INPUT1>',
+        help="Input FASTQ/FASTA file. Assumes to contain read1 if given with second input file. If only "
+             "input1 is given, input is assumed to be an interleaved. If reading from stdin"
+             "is requested use '-' as a placeholder.")
+    parser.add_argument(
+        "input2", nargs='?', metavar='<INPUT2>',
+        help="Input  FASTQ/FASTA for read2 for paired-end read. Leave empty if using interleaved.")
+    parser.add_argument(
+        "-o1", "--output1", default=None, metavar='<OUTPUT1>',
```

Comment:
Single dashes should generally only be used with single-character command-line options. (This is a convention established by `getopt_long`.) The reason is that you can merge multiple single-character options that don’t require arguments into one: `ls -la` instead of `ls -l -a`. This gets ambiguous as soon as you allow multi-character options with a single dash. argparse somehow handles this, but I would not rely on it. Cutadapt uses `-o` and `-p` because of this. You could also use `--o1` instead.

Same comment applies to options below.

Reply:
> Regarding the extra barcode extraction step, I don’t think avoiding this is easily possible with Cutadapt at the moment. Also, I don’t have a good idea for a general syntax for this on the command-line, so I would even know how to start.
> 
> However, if the extra runtime becomes sufficiently annoying, I can look into using Cutadapt directly as a Python module. It doesn’t have an official API, but if we pin the version, this shouldn’t be a problem.

Then we will keep this for now and change it only if it becomes necessary. 

I have include (hopefully) all changes now. Thank you for your thorough reviews!!